workspace_status.py

```diff
@@ -0,0 +1,69 @@
+# Get the current repository git status.
+# This means get the current version tag and if the repository is dirty.
+import subprocess
+import sys
+import argparse
+
+
+def main():
+    parser = argparse.ArgumentParser(description='')
+    parser.add_argument('--default_version',
+                        required=True,
+                        help='default version in case git can not be called')
+    args = parser.parse_args()
+
+    # Get the current status of the repository by calling out to git.
+    # In case there is no git executable use the default version.
+    git_version = get_version(".")
+    git_is_dirty = get_git_dirty(".")
+
+    # Write to volatile-status.txt.
+    # This is a bazel thing and the recommended way of
+    # getting version control status into bazel build according
+    # to bazels documentation.
+    print("GIT_VERSION {}".format(git_version))
```

Comment:
how does this write to volatile-status.txt? it just outputs to stdout no?

Reply:
https://docs.bazel.build/versions/main/user-manual.html
--> 
```
--workspace_status_command=program

This flag lets you specify a binary that Bazel runs before each build. The program can report information about the status of the workspace, such as the current source control revision.

The flag's value must be a path to a native program. On Linux/macOS this may be any executable. On Windows this must be a native binary, typically an ".exe", ".bat", or a ".cmd" file.

The program should print zero or more key/value pairs to standard output, one entry on each line, then exit with zero (otherwise the build fails). The key names can be anything but they may only use upper case letters and underscores. The first space after the key name separates it from the value. The value is the rest of the line (including additional whitespaces). Neither the key nor the value may span multiple lines. Keys must not be duplicated.

Bazel partitions the keys into two buckets: "stable" and "volatile". (The names "stable" and "volatile" are a bit counter-intuitive, so don't think much about them.)

**Bazel then writes the key-value pairs into two files:**

    **bazel-out/stable-status.txt contains all keys and values where the key's name starts with STABLE_
    bazel-out/volatile-status.txt contains the rest of the keys and their values**

```

Reply:
TLDR; Its a bazel thing. everything to stdout in the workspace_status_command ends up in the volatile-status.txt or stable-status.text

Reply:
ah thanks. i would have accepted "rtfm" too :)